__tests__/gui/rich_data/rich_data.js

```diff
@@ -0,0 +1,37 @@
+import {When, Given, Then} from "cypress-cucumber-preprocessor/steps";
+import all_details from "../rich_data/all_details.json";
+
+Given('I am on the Wazimap Homepage', () => {
+    cy.visit("/")
+
+    cy.intercept('/api/v1/all_details/profile/8/geography/ZA/?format=json', (req) => {
+        req.reply({
+            statusCode: 201,
+            body: all_details,
+            forceNetworkError: false // default
+        })
+    })
+})
+
+When('I expand Rich Data', () => {
+    cy.wait(100);
+    cy.get('.point-mapper-toggles .rich-data-panel__open').click();
+})
+
+Then('Rich Data should be displayed', () => {
+    cy.get('.rich-data-nav__list').should('be.visible');
+})
+
+When('I mouseover the hamburger menu', () => {
+    cy.get('.hover-menu').first().trigger('mouseover');
+})
+
+Then('the hamburger menu should show', () => {
+    cy.get('.hover-menu__content').should('be.visible');
+})
+
+When('I click on Save As Image in Rich Data', () => {
+    cy.get('.hover-menu__content_item').first().click();
+    cy.verifyDownload('Youth status.png');
```

Comment:
```suggestion
import {waitUntilGeographyIsLoaded} from "../common_cy_functions/general";
import all_details from "../rich_data/all_details.json";

Given('I am on the Wazimap Homepage', () => {
    cy.visit("/")

    cy.intercept('/api/v1/all_details/profile/8/geography/ZA/?format=json', (req) => {
        req.reply({
            statusCode: 201,
            body: all_details,
            forceNetworkError: false // default
        })
    })
})

Then('I wait until map is ready', () => {
    waitUntilGeographyIsLoaded('South Africa');
})

When('I expand Rich Data', () => {
    cy.wait(100);
    cy.get('.point-mapper-toggles .rich-data-panel__open').click();
})

Then('Rich Data should be displayed', () => {
    cy.get('.rich-data-nav__list').should('be.visible');
})

When('I mouseover the hamburger menu', () => {
    cy.get('.hover-menu').first().trigger('mouseover');
})

Then('the hamburger menu should show', () => {
    cy.get('.hover-menu__content').should('be.visible');
})

When('I click on Save As Image in Rich Data', () => {
    cy.get('.hover-menu__content_item').first().click();
    cy.verifyDownload('Total population by age group.png');
```

Reply:
Hoped suggested change would show differences better. Basically the test was missing a method for 'I wait until map is ready' and the correct name for the downloaded file. That should fix CI

Reply:
nice catch @michaelglenister - apart from this I needed to add some json files for the test too. FYI